examples/sbol2/CreatingSBOL2Objects/Component.ipynb

```diff
@@ -0,0 +1,195 @@
+    "plasmid_sequence.encoding = sbol2.SBOL_ENCODING_IUPAC\n",
+    "doc.addSequence(plasmid_sequence)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ceate a `ComponentDefinition` object for the promoter `Sequence`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 65,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plasmid_component_definition = sbol2.ComponentDefinition(uri='plasmid_component_definition', component_type=sbol2.BIOPAX_DNA)\n",
+    "plasmid_component_definition.addRole(sbol2.SO_PROMOTER)\n",
+    "plasmid_component_definition.sequence = plasmid_sequence\n",
+    "doc.addComponentDefinition(plasmid_component_definition)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, Create a `Range` object to define the portion of the plasmids's sequence that will be included in the RBS (lets say 15 - 28)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 66,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "promoter_range = sbol2.Range(uri='sample_range')\n",
+    "promoter_range.start = 15\n",
+    "promoter_range.end = 28\n",
+    "doc.add(promoter_range)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create a `Component` object for the Promoter that is situated inside the plasmid"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 68,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "promoter_component = sbol2.Component(uri='RBS_component', access=sbol2.SBOL_ACCESS_PRIVATE)\n",
+    "promoter_component.definition = plasmid_component_definition\n",
+    "promoter_component.sourceLocations = [promoter_range]\n",
+    "promoter_component.roles = [sbol2.SO_PROMOTER]\n",
+    "promoter_component.roleIntegration = sbol2.SBOL_ROLE_INTEGRATION_OVERRIDE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ceate a `ComponentDefinition` object for the Gene the Promoter will be promoting, and then add the promoter to said compo. You can optionally add a `Sequence` and `SequenceAnnotation` to the Gene to specify the full gene sequence as well as where the promoter for the gene is situated, however we will not go into how to do that here. For more information, go over to the relevent NoteBooks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 67,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "example_gene_component_definition = sbol2.ComponentDefinition(uri='example_gene_component_definition', component_type=sbol2.BIOPAX_DNA)\n",
+    "example_gene_component_definition.addRole(sbol2.SO_GENE)\n",
+    "example_gene_component_definition.addRole(sbol2.SO_PROMOTER)\n",
+    "doc.addComponentDefinition(example_gene_component_definition)\n",
+    "example_gene_component_definition.components.add(promoter_component)"
```

Comment:
If you put in a CDS and an ordering constraint, then you can also demonstrate how pySBOL2 lets you calculate a sequence for the containing `ComponentDefinition`, using the `ComponentDefinition.compile` method.

Reply:
I feel that this functionality can be shown off in the sequence_constraint notebook. Since that is where sequence constraints are going to be shown off in full

Reply:
That's reasonable if you want to keep this simple.
I would suggest leaving a comment in here that points to that notebook, though.